File: akka-samples/akka-sample-main-scala/src/main/scala/sample/hello/HelloWorld.scala
/**
 *  Copyright (C) 2009-2013 Typesafe Inc. <http://www.typesafe.com>
 */
package sample.hello

import akka.actor.Actor
import akka.actor.Props

class HelloWorld extends Actor {

  override def preStart(): Unit = {
    // create the greeter actor
    val greeter = context.actorOf(Props[Greeter], "greeter")
    // tell it to perform the greeting
    greeter ! Greeter.Greet
  }

  def receive = {
    // when the greeter is done, stop this actor and with it the application
    case Greeter.Done ⇒ context.stop(self)
  }
}

File: akka-samples/akka-sample-main-java-lambda/src/main/java/sample/hello/HelloWorld.java
package sample.hello;

import akka.actor.AbstractActor;
import akka.actor.Props;
import akka.actor.ActorRef;
import akka.japi.pf.ReceiveBuilder;
import static sample.hello.Greeter.Msg;

public class HelloWorld extends AbstractActor {

  public HelloWorld() {
    receive(ReceiveBuilder.
      matchEquals(Msg.DONE, m -> {
        // when the greeter is done, stop this actor and with it the application
        context().stop(self());
      }).build());
  }

  @Override
  public void preStart() {
    // create the greeter actor
    final ActorRef greeter = getContext().actorOf(Props.create(Greeter.class), "greeter");
    // tell it to perform the greeting
    greeter.tell(Greeter.Msg.GREET, self());
  }
}
